src/grammar/clone.ts

```diff
@@ -28,7 +28,6 @@ export function cloneNode<T extends ASTNode>(src: T): T {
     } else if (src.kind === "statement_let") {
         return cloneASTNode({
             ...src,
-            type: cloneASTNode(src.type),
```

Comment:
you are sure we won't run into issues in the future if we just ignore the type here?

for instance, `def_argument` has the following:
```
    } else if (src.kind === "def_argument") {
        return cloneASTNode({
            ...src,
            type: cloneASTNode(src.type),
        });
```

I'd say we should either clone all types or not clone any.

Reply:
those aren't cloned in `ASTField`, `ASTConstant`, `ASTArgument`, `ASTFunction` and `ASTNativeFunction` too. Should I add that type cloning to all of these?

Reply:
good point, let's clone them all